executor/executor.go

```diff
@@ -78,7 +78,8 @@ type baseExecutor struct {
 	maxChunkSize  int
 	children      []Executor
 	retFieldTypes []*types.FieldType
-	runtimeStat   *execdetails.RuntimeStat
+	statsEnable   bool
+	_runtimeStats *execdetails.RuntimeStats
```

Comment:
I think the old way is better. And to eliminate misunderstanding that `e.runtimeStats` will always be set, in function `newBaseExecutor()` we can:
```go
if ctx.GetSessionVars().StmtCtx.RuntimeStatsColl != nil {
	e.runtimeStats = e.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.GetRuntimeStat(e.id)
}
```

Reply:
Done~